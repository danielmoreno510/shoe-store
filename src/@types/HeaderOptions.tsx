type HeaderOptionsProps = {
  badge?: number;
  changeScreen(): any;
  position: string;
};
