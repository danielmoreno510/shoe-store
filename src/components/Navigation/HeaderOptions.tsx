import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import {Icon, Badge} from 'native-base';

import styles from './styles';

const HeaderOptions: React.FC<HeaderOptionsProps> = ({badge = 0, changeScreen, position}) => (
  <TouchableOpacity
    onPress={changeScreen}
    style={[position === 'left' ? styles.headerLeftContainer : styles.headerRightContainer]}>
    {!!badge && (
      <Badge warning style={styles.badge}>
        <Text style={styles.badgeText}>{badge}</Text>
      </Badge>
    )}
    <Icon
      type="MaterialIcons"
      name={position === 'left' ? 'menu' : 'shopping-cart'}
      style={styles.iconColor}
    />
  </TouchableOpacity>
);

export default HeaderOptions;
