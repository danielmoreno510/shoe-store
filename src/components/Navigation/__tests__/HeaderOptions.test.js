import React from 'react';
import {shallow} from 'enzyme';

import HeaderOptions from '../HeaderOptions';

describe('HeaderOptions component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      changeScreen: jest.fn(),
      position: 'left',
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<HeaderOptions {...testProps} />);

      it('THEN should display a regular HeaderOptions', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('WHEN the component is rendered with badge', () => {
      const customProps = {...testProps, badge: 1, position: 'right'};
      const shallowWrapper = shallow(<HeaderOptions {...customProps} />);

      it('THEN should display a regular HeaderOptions', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
