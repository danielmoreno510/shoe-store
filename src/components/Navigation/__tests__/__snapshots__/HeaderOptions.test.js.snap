// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderOptions component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular HeaderOptions 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "marginLeft": 12,
      },
    ]
  }
>
  <Component
    name="menu"
    style={
      Object {
        "color": "#f1b30a",
      }
    }
    type="MaterialIcons"
  />
</ForwardRef>
`;

exports[`HeaderOptions component GIVEN a set of valid props WHEN the component is rendered with badge THEN should display a regular HeaderOptions 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "marginRight": 12,
      },
    ]
  }
>
  <Component
    style={
      Object {
        "height": 22,
        "marginBottom": -12,
        "marginLeft": 14,
        "width": 22,
        "zIndex": 1,
      }
    }
    warning={true}
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      1
    </Text>
  </Component>
  <Component
    name="shopping-cart"
    style={
      Object {
        "color": "#f1b30a",
      }
    }
    type="MaterialIcons"
  />
</ForwardRef>
`;
